cli/pkg/up/up.go

```diff
@@ -13,23 +17,43 @@ import (
 	template2 "github.com/rancher/rio/pkg/template"
 )
 
-func readFileInPath(cwd, file string) ([]byte, error) {
+func readFileInPath(cwd, file, filePath string) ([]byte, error) {
+	if strings.HasPrefix(filePath, "http") {
+		base, err := url.Parse(filePath)
+		if err != nil {
+			return nil, err
+		}
+		ref, err := url.Parse(filepath.Join(file))
+		if err != nil {
+			return nil, err
+		}
+		resolved := base.ResolveReference(ref)
+		if err != nil {
+			return nil, err
+		}
+		resp, err := http.Get(resolved.String())
+		if err != nil {
+			return nil, err
+		}
+		defer resp.Body.Close()
+		return ioutil.ReadAll(resp.Body)
+	}
 	f, err := symlink.FollowSymlinkInScope(file, cwd)
 	if err != nil {
 		return nil, err
 	}
 	return ioutil.ReadFile(f)
 }
 
-func readFiles(files []string, template *template2.Template, promptReplaceFile bool) error {
+func readFiles(filePath string, files []string, template *template2.Template, promptReplaceFile bool) error {
 	cwd, err := os.Getwd()
```

Comment:
remove this getwd code and have the caller pass in the "relativeTo" which is either `http` url or a assumed to be a local directory